app/views/settings/preferences/show.html.erb

```diff
@@ -42,27 +42,31 @@
 <% end %>
 
 <%= settings_section title: t(".theme_title"), subtitle: t(".theme_subtitle") do %>
-  <div>
-    <%= styled_form_with model: @user, class: "flex flex-col md:flex-row justify-between items-center gap-4", data: { controller: "auto-submit-form" } do |form| %>
```

Comment:
Just to stay consistent with the rest of the codebase, I think we probably want to keep this existing auto-submit controller wired up here.

I agree with your idea of wrapping those radio buttons though so clicks can be captured anywhere on the image or radio input.  

Can we find a way to re-structure the HTML with a form label of some sort?

I think changing from `styled_form_with` to a regular `form_with` to avoid all those default form styles might help quite a bit here and give you more access to the inputs directly.

Reply:
This was great feedback thanks! I changed a bunch of stuff up here - it's now using the `form_with` and html form labels. I've removed the theme specific controller as well so that's nice. I also swapped some other stuff around to remove a bunch of code and iterate over a list. As a part of all of this, the performance of all of it is nearly instant and it feels very smooth:

https://github.com/user-attachments/assets/85612d83-efe9-490f-affc-f30df332beda

Reply:
Looks great!